rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -143,9 +143,10 @@ bool SequentialReader::has_next()
     if (!current_storage_has_next && has_next_file()) {
       load_next_file();
       storage_options_.uri = get_current_file();
-
+      // open and set filters
       storage_ = storage_factory_->open_read_only(storage_options_);
-      storage_->set_filter(topics_filter_);
+      storage_->seek(seek_time_);
```

Comment:
Would it make sense for this `if`-block logic to go into `load_next_file` instead of being here in `has_next`?